src/pages/iou/IOUTransactions.js

```diff
@@ -0,0 +1,63 @@
+import React, {PureComponent} from 'react';
+import {View} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import _ from 'underscore';
+import PropTypes from 'prop-types';
+import styles from '../../styles/styles';
+import ONYXKEYS from '../../ONYXKEYS';
+import ReportActionPropTypes from '../home/report/ReportActionPropTypes';
+import IOUTansactionPropTypes from './IOUTansactionPropTypes';
+import ReportTransaction from '../../components/ReportTransaction';
+
+const propTypes = {
+    // Actions from the ChatReport
+    reportActions: PropTypes.shape(ReportActionPropTypes),
+
+    // ReportID for the associated chat report
+    chatReportID: PropTypes.number.isRequired,
+
+    // ReportID for the associated IOU report
+    iouReportID: PropTypes.number.isRequired,
+
+    // Transactions for this IOU report
+    transactions: PropTypes.arrayOf(PropTypes.shape(IOUTansactionPropTypes)),
+};
+
+const defaultProps = {
+    reportActions: {},
+    transactions: [],
+};
+
+class IOUTransactions extends PureComponent {
+    render() {
+        return (
+            <View style={[styles.mt3]}>
+                {_.map(this.props.transactions, (transaction) => {
```

Comment:
> I'm confused here, are we rendering a ReportTransaction for every transaction?
Yes. One Component for each transaction. 

This logic is **matching an IOU report transaction to the associated ChatReport action** so that we can pass both to ReportTransaction. This is because we currently use IOU transactions as the data source, but they lack necessary information that is retrieved via the chat report action (the message).

Is this bad and should it be changed? Yes, but I've separated this task out into the follow-up issue ('Implement Cancel/decline transactions within IOU Details'). Rejection of transactions and displaying 'Settled|Cancelled|Declined' events in the IOU details Modal also depends on this change, so it will be done.

Reply:
Ok, but to clarify, it's the JS syntax that we're using that confuses me.

I see use of `map()` which implies that for every "transaction" in `props.transactions` we are returning a `ReportTransaction`. Can `actionForTransaction` be `undefined` and yet we render a `ReportTransaction` anyway? Or is that not possible? Can we maybe add a comment in the code to explain what we're doing here?

Reply:
Fiar point. I think I've cleared up and clarified this logic now. [Here's the commit](https://github.com/Expensify/Expensify.cash/pull/2394/commits/4ae8b8528dd436207cd7baf2a7bd81b92511ded7).

Reply:
But that's correct, we should never not find a matching action and we will now throw an error if this does occur.

Reply:
Ok I think the extra comments help here. I have some other questions but I'll start a new thread.